util: improves error messages on get_previous_releases script

NelsonGaldeman opened the pull request:
When previous releases are fetched and the specified version wasn't added to the checksum list we used to get a "Checksum did not match" which isn't true (https://github.com/bitcoin-core/bitcoincore.org/issues/753#issuecomment-879546719). 

If the specified version number is not on the list, it now logs cannot do the comparison instead.

theStack commented:
Concept ACK, will test later

rajarshimaitra reviewed (commented):
Concept ACK with an error message comment.

NelsonGaldeman commented on test/get_previous_releases.py:
```diff
@@ -104,7 +104,12 @@ def download_binary(tag, args) -> int:
+            print("Checksum did not match")
+            return 1
+
+        print("Cannot compare checksum for version {version}".format(
```
Agree, my wording wasn't the best! Thank you for the suggestion. Updated.

theStack commented on test/get_previous_releases.py:
```diff
@@ -119,6 +123,13 @@ def download_binary(tag, args) -> int:
 
 
+def checksum_exists(version) -> bool:
+    for sum in SHA256_SUMS:
+        if "bitcoin-" + version in SHA256_SUMS[sum]:
+            return True
+    return False
```
I think this could be simplified to a one-liner (maybe it could directly be used above then instead of introducing an extra function):
```suggestion
    return "bitcoin-" + version in SHA256_SUMS.values()
```

theStack replied:
Looking a second time at it, that was too simple-thought; it should be more like
```
    return any("bitcoin-" + version in str for str in SHA256_SUMS.values())
```

NelsonGaldeman replied:
Yes absolutely! My python is a little bit rusty it's been a while. I didn't know I could do that, it looks cleaner. Thank you!

theStack commented on test/get_previous_releases.py:
```diff
@@ -104,7 +104,11 @@ def download_binary(tag, args) -> int:
 
     if tarballHash not in SHA256_SUMS or SHA256_SUMS[tarballHash] != tarball:
-        print("Checksum did not match")
+        if checksum_exists(tag[1:]):
```
nit: could also pass the full filename (`tarball`) here, then the function `checksum_exists` below wouldn't need to add the "bitcoin-" prefix and is more specific?

NelsonGaldeman replied:
I purposely avoided passing the full name in case the naming convention changes it will still work. Thinking more about it would be helpful to check the full name as it will throw the checksum doesn't exist if we are missing a specific OS/arch on the list even if we included some for that same version.

theStack reviewed (commented):
Left two possible improvements that came up while reviewing the code. Couldn't test yet (the machine I am using right now is running OpenBSD, where the script doesn't work; even if the platform is overrided to e.g. 'x86_64-linux-gnu', the tar command fails), but will do tomorrow on a Linux machine.

fanquake commented:
> even if the platform is overrided to e.g. 'x86_64-linux-gnu', the tar command fails),

That sounds like something we should be able to fixup.

NelsonGaldeman committed:
util: improves error messages on get_previous_releases script

If a requested version doesn't exist on our list of checksums it says it cannot do a checksum comparision instead of saying it did not match

theStack reviewed (approved):
tACK 179a051704321ba40277a5855d6ac0dbb45689dd, tested on Debian bullseye/sid

master branch:
```
$ ./test/get_previous_releases.py -b v0.21.0
Releases directory: releases
Fetching: https://bitcoincore.org/bin/bitcoin-core-0.21.0/bitcoin-0.21.0-x86_64-linux-gnu.tar.gz
[ ... ]
Checksum did not match
$ echo $?
1
```
PR branch:
```
$ ./test/get_previous_releases.py -b v0.21.0
Releases directory: releases
Fetching: https://bitcoincore.org/bin/bitcoin-core-0.21.0/bitcoin-0.21.0-x86_64-linux-gnu.tar.gz
[ ... ]
Checksum for given version doesn't exist
$ echo $?
1
```

By the way, welcome as a new contributor NelsonGaldeman! 🎉 

> > even if the platform is overrided to e.g. 'x86_64-linux-gnu', the tar command fails),
> 
> That sounds like something we should be able to fixup.

Yes, will take a look into that.

practicalswift commented:
cr ACK 179a051704321ba40277a5855d6ac0dbb45689dd

Warm welcome as a new contributor @NelsonGaldeman! 🥇
